src/app/pages/storage/snapshots/snapshot-list/snapshot-list.component.ts

```diff
@@ -273,8 +279,52 @@ export class SnapshotListComponent {
     });
   }
 
+  restructureData(selected: SnapshotData[]): DialogData {
+    const datasets: string[] = [];
+    const snapshots: any = {};
```

Comment:
You can add `{ [dataset: string]: SnapshotData }` here and in DialogData.

Reply:
Done